modyn/trainer_server/internal/trainer/remote_downsamplers/abstract_matrix_downsampling_strategy.py

```diff
@@ -0,0 +1,110 @@
+from abc import abstractmethod
+from enum import Enum
+from typing import Any, Optional
+
+import numpy as np
+import torch
+from modyn.trainer_server.internal.trainer.remote_downsamplers.abstract_per_label_remote_downsample_strategy import (
+    AbstractPerLabelRemoteDownsamplingStrategy,
+)
+from modyn.trainer_server.internal.trainer.remote_downsamplers.deepcore_utils.shuffling import _shuffle_list_and_tensor
+
+MatrixContent = Enum("MatrixContent", ["EMBEDDINGS", "GRADIENTS"])
+
+
+class AbstractMatrixDownsamplingStrategy(AbstractPerLabelRemoteDownsamplingStrategy):
```

Comment:
This class is used to abstract the common behaviour of Kcentergreedy, Submodular and GradMatch